dask/array/core.py

```diff
@@ -3838,6 +3838,21 @@ def repeat(a, repeats, axis=None):
     return concatenate(out, axis=axis)
 
 
+@wraps(np.tile)
+def tile(A, reps):
+    if not isinstance(reps, Integral):
+        raise NotImplementedError("Only integer valued `reps` supported.")
```

Comment:
I think I'm going to hold off on implementing this for now. There's a lot of edge cases to consider and I'm not ready to take them on yet. It's certainly solvable just as it is with `repeat`.

That said, I'm not a huge fan of Python's recursion. It's not terribly efficient and Python has better tools to tackle these problems thanks to iterators. So if (or when) I might implement this, I'd probably go with some sort of `for`-loop construction.